### ISSUE TITLE
Error when logging into Google Docs

### ISSUE THREAD

**Issue by ogomez92** [cypress-io/cypress#22882]
### Current behavior

Currently, Cypress is not letting me login using the Google Docs login form. It displays an error on the page with no additional message, not even showing me as a bot.

We have a Google Add-on that we want to test, and for this we need to login to Google Docs, I already know about the API based authentication for specific apps but I don't need a token, I need to login to Google.

I have tried several things, including passing my Chrome profile as --user-data-dir and preserving its cookies (didn't work), and also to login using docs.google.form sign in form. as soon as you hit next on this form, it breaks with a generic error message.

I also tried setting the user agent, and using another browser like Edge .

### Desired behavior

It should let you login normally and show the password field, focus it, and let you type in the password for login.

### Test code to reproduce

```js
module.exports = defineConfig({
  "chromeWebSecurity": false,
  e2e: {
    "chromeWebSecurity": false,
    baseUrl: 'https://docs.google.com',
    experimentalSessionAndOrigin: true,
  }
})
```

// Test code

```js
describe('empty spec', () => {
  it('opens the google test document', () => {
    cy.visit('/', {
      headers: {
        'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36',
      }
    });

    googleLoginPage.getEmailAddressField().type(Cypress.env('googleUser'));
    googleLoginPage.getNextButton().click()

  })
})
```

### Cypress Version

10.3.1

### Other

Not sure other that I tried a lot of different things and none works.. :(

**Comment by ogomez92** [comment]
Forgot to include google login page!

```Javascript
class GoogleLogin {
    getEmailAddressField() {
        return cy.get('#identifierId');
    }

    getNextButton() {
        return cy.get('#identifierNext');
    }
}

export default GoogleLogin;

```

**Comment by ogomez92** [comment]
Here is the request:
https://www.oriolgomez.com/files/request.har

**Comment by mjhenkes** [comment]
@ogomez92, I'm sorry that you're having this issue.  We recently have looked into enabling cypress with google auth for issue https://github.com/cypress-io/cypress/issues/21474 and have spawned a couple of other tasks to try and improve the problem (#22952 and #22953). With these fixes we were able to log into google via edge, but not through chrome as there seems to be more security in place around google auth in googles own browser. We're doing our best, but at the end of the day, google wants to prevent automated logins to their auth system.

**Comment by ogomez92** [comment]
I see this issue was apparently solved in version 10.4.0 https://github.com/cypress-io/cypress/issues/21474

I looked at the released notes and saw the new flag, I went ahead and enabled this in cypress.config
```json
module.exports = defineConfig({

  e2e: {
    "chromeWebSecurity": false,
    baseUrl: 'https://docs.google.com',
    experimentalSessionAndOrigin: true,
    experimentalModifyObstructiveThirdPartyCode : true,
..
```

Then I tried to login using the same test code I posted at first.
Am I doing something wrong do I need to enable something else for it to work?

I'm using Electron because I see that Google Chrome is not yet supported.

Thanks.

**Comment by AtofStryker** [comment]
@ogomez92 seems like a legit bug. I created a [repository](https://github.com/AtofStryker/cypress-issue-22882) that recreates the issue

**Comment by AtofStryker** [comment]
possibly fixed with #23885 if using `cy.origin` and #24094 if trying to interact with `google` as top

**Comment by ogomez92** [comment]
Great, thanks!
